src/components/patch/PatchContainer.tsx

```diff
@@ -108,37 +192,59 @@ export class PatchContainer extends React.Component<Props, State> {
   }
 
   private filterItems(query: string) {
-    const filtered: UIVersion[] = [];
+    const filtered: UIPatch[] = [];
     if (query === "") {
       this.setState({
         isSearching: false
       });
       return this.state.allPatches;
     }
     Object.keys(this.state.allPatches).map(versionId => {
-      const patch = this.state.allPatches[versionId];
-      const description = patch.Version.message;
-      if (description.toLowerCase().indexOf(query.toLowerCase()) !== -1) {
+      const patch = this.state.allPatches[versionId] as UIPatch;
+      const description = patch.Patch.Description;
+      const status = patch.Patch.Status;
+      const project = patch.Patch.Project;
+      if ((this.state.selectedProjects.length === 0 || this.state.selectedProjects.indexOf(project) > -1) &&
+        (this.state.selectedStatuses.length === 0 || this.state.selectedStatuses.indexOf(status) > -1) &&
+        (description.toLowerCase().indexOf(query.toLowerCase()) !== -1)) {
         filtered.push(patch);
       }
     });
     return filtered;
   }
 
-  private updateOpenPatches = (patchObj: UIVersion) => {
+  private updateOpenPatches = (patchObj: UIPatch) => {
     const newExpanded = this.state.expandedPatches;
-    if (patchObj.Version.id in this.state.expandedPatches) {
-      delete newExpanded[patchObj.Version.id];
+    if (patchObj.Patch.Id in this.state.expandedPatches) {
+      delete newExpanded[patchObj.Patch.Id];
     } else {
-      newExpanded[patchObj.Version.id] = 1;
+      newExpanded[patchObj.Patch.Id] = 1;
     }
     this.setState({
       expandedPatches: newExpanded
     }, this.props.onFinishStateUpdate);
   }
 
-  private isExpanded = (patchObj: UIVersion) => {
-    return patchObj.Version.id in this.state.expandedPatches;
+  private isExpanded = (patchObj: UIPatch) => {
+    return patchObj.Patch.Id in this.state.expandedPatches;
+  }
+
+  private onStatusSelectChange = (event: React.ChangeEvent<{ name?: string; value: string }>) => {
+    const selectedValues = event.target.value as unknown as string[];
```

Comment:
Do you need to case twice for this to work?

Reply:
Yes- if not, you get the following error: "Conversion of type 'string' to type 'string[]' may be a mistake because neither type sufficiently overlaps with the other. If this was intentional, convert the expression to 'unknown' first."